src/App.tsx

```diff
@@ -1,31 +1,108 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  FC,
+  useContext,
+  useState,
+  useEffect,
+  useReducer,
+  useMemo,
+  useRef,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification }
+  from './components/ErrorNotidication/ErrorNotification';
+import { TodoList } from './components/ToDoList/ToDoList';
+import { Footer } from './components/Footer/Footer';
+
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { SortType } from './types/Filter';
+
+function filtTodos(
+  todos: Todo[],
+  sortType: SortType,
+) {
+  const visibleTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      return visibleTodos.filter(todo => !todo.completed);
+
+    case SortType.Completed:
+      return visibleTodos.filter(todo => todo.completed);
+    default:
+      return visibleTodos;
+  }
+}
+
+const reducer = (count: number, action: string) => {
+  switch (action) {
+    case 'increase':
+      return count + 1;
+    case 'decrease':
+      return count - 1;
+    case 'clear':
+      return 0;
+    default:
+      return count;
+  }
+};
 
-export const App: React.FC = () => {
+export const App: FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [sortType, setSortType] = useState<SortType>(SortType.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completeItem, dispatch] = useReducer(reducer, 0);
+
+  const increase = () => dispatch('increase');
+
+  const getUserFromServer = (userId: number) => {
+    getTodos(userId)
+      .then(userTodosFromServer => setTodos(userTodosFromServer))
+      .catch(() => setErrorMessage('Unable to update todos'));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code